descartes_planner/src/sparse_planner.cpp

```diff
@@ -648,29 +686,53 @@ bool SparsePlanner::plan()
       case int(InterpolationResult::REPLAN):
           replan = true;
           cart_point = cart_points_[point_pos];
-
           if(sparse_index == 0)
           {
+            // If the point is being inserted at the beginning of the trajectory
+            // there is no need to tweak the timing that comes from the dense traj
             prev_id = descartes_core::TrajectoryID::make_nil();
             next_id = std::get<1>(sparse_solution_array_[sparse_index])->getID();
           }
           else
           {
+            // Here we want to calculate the time from the prev point to the new point
+            int prev_dense_id = std::get<0>(sparse_solution_array_[sparse_index-1]);
+            int next_dense_id = point_pos;
+            descartes_core::TimingConstraint tm = cumulativeTimingBetween(cart_points_, prev_dense_id, next_dense_id);
+            descartes_core::TrajectoryPtPtr copy_pt = copyAndSetTiming(cart_point, tm);
+            cart_point = copy_pt; // swap the point over
+
             prev_id = std::get<1>(sparse_solution_array_[sparse_index-1])->getID();
             next_id = std::get<1>(sparse_solution_array_[sparse_index])->getID();
           }
 
-          if(planning_graph_->addTrajectory(cart_point,prev_id,next_id))
+          // In either case, the sparse_index point will have to be recalculated
           {
-            sparse_solution_array_.clear();
-            ROS_INFO_STREAM("Added new point to sparse trajectory from dense trajectory at position "<<
-                            point_pos<<", re-planning entire trajectory");
-          }
-          else
-          {
-            ROS_ERROR_STREAM("Adding point "<<point_pos <<"to sparse trajectory failed, aborting");
-            replan = false;
-            succeeded = false;
+            int prev_dense_id = point_pos;
+            int next_dense_id = std::get<0>(sparse_solution_array_[sparse_index]);
+            descartes_core::TimingConstraint tm = cumulativeTimingBetween(cart_points_, prev_dense_id, next_dense_id);
+            descartes_core::TrajectoryPtPtr copy_pt = copyAndSetTiming(cart_points_[next_dense_id], tm);
+
+            if (!planning_graph_->modifyTrajectory(copy_pt))
+            {
+              ROS_ERROR_STREAM("Could not modify trajectory point with id: " << copy_pt->getID());
+              replan = false;
+              succeeded = false;
```

Comment:
+1 If we run into this situation then we're modifying a point in the sparse trajectory that is NOT in the planning_graph. This should never happen, but sanity checks are good.